src/main/java/net/imagej/ops/image/scale/ScaleImg.java

```diff
@@ -52,56 +55,75 @@
  */
 @Plugin(type = Ops.Image.Scale.class)
 public class ScaleImg<T extends RealType<T>> extends
-	AbstractUnaryFunctionOp<Img<T>, Img<T>> implements Ops.Image.Scale
+	AbstractUnaryFunctionOp<Img<T>, Img<T>> implements Ops.Image.Scale,
+	Contingent
 {
 
 	@Parameter
 	/*Scale factors for each dimension*/
 	private double[] scaleFactors;
 
+	private ImgFactory<T> factory;
+
+	private UnaryComputerOp<IntervalView<T>, Img<T>> copyImg;
+
 	@Parameter
 	private InterpolatorFactory<T, RandomAccessible<T>> interpolator;
 
 	@Override
-	public Img<T> compute1(Img<T> input) {
-		if (input.numDimensions() != scaleFactors.length) {
-			throw new IllegalArgumentException(
-				"Less/more scale factors are provided than dimensions input the image.");
+	public boolean conforms() {
+		return in().numDimensions() == scaleFactors.length;
+	}
+
+	@SuppressWarnings({ "unchecked", "rawtypes" })
+	@Override
+	public void initialize() {
+		final long[] newDims = new long[in().numDimensions()];
+		in().dimensions(newDims);
+		for (int i = 0; i < Math.min(scaleFactors.length, in()
+			.numDimensions()); i++)
+		{
+			newDims[i] = Math.round(in().dimension(i) * scaleFactors[i]);
+		}
+
+		if (in().firstElement() instanceof NativeType) {
+			@SuppressWarnings("rawtypes")
+			final NativeType type = (NativeType) in().firstElement().createVariable();
+			factory = ops().create().imgFactory(new FinalDimensions(newDims), type);
+		}
+		else {
+			factory = in().factory();
 		}
 
-		final long[] newDims = new long[input.numDimensions()];
-		input.dimensions(newDims);
-		for (int i = 0; i < Math.min(scaleFactors.length, input
+		IntervalView<T> interval = Views.interval(Views.raster(RealViews.affineReal(
+			Views.interpolate(Views.extendMirrorSingle(in()), interpolator),
+			new net.imglib2.realtransform.Scale(scaleFactors))), new FinalInterval(
+				newDims));
+
+		copyImg = (UnaryComputerOp) Computers.unary(ops(),
+			Ops.Copy.IterableInterval.class, Img.class, interval);
+	}
+
+	@Override
+	public Img<T> compute1(Img<T> input) {
+		final long[] newDims = new long[in().numDimensions()];
```

Comment:
Whoa. The `in()` method should never be called from `compute1`. Why are we not using `input` instead here?